src/toxcore/dht/lan_discovery.rs

```diff
@@ -128,11 +196,11 @@ impl LanDiscoverySender {
 
     /// Run LAN discovery periodically. Result future will never be completed
     /// successfully.
-    pub fn run(mut self) -> IoFuture<()> {
+    pub fn run(mut self) -> impl Future<Item=(), Error=LanDiscoveryError> + Send {
```

Comment:
Do we need to wrap errors when there is only one possible error (`tokio::timer::Error` here)? @kpp ?